igvc_navigation/src/mapper/mapper.cpp

```diff
@@ -142,25 +148,43 @@ void Mapper::insertCameraProjection(const pcl::PointCloud<pcl::PointXYZ>::ConstP
 }
 
 void Mapper::insertSegmentedImage(cv::Mat&& image, const tf::Transform& base_to_odom,
-                                  const tf::Transform& camera_to_base, const ros::Time& stamp)
+                                  const tf::Transform& camera_to_base, const ros::Time& stamp, Camera camera)
 {
+  if (camera_model_map_.find(camera) == camera_model_map_.end() && !camera_model_initialized_ && !use_ground_filter_)
+  {
+    ROS_WARN_STREAM_THROTTLE(1, "camera model not initialized or not using ground filter");
+    return;
+  }
+
   PointCloud projected_pc;
   processImageFreeSpace(image);
 
-  if (!camera_model_initialized_ && !use_ground_filter_)
+  image_geometry::PinholeCameraModel model = camera_model_map_.at(camera);
+
+  MapUtils::projectToPlane(projected_pc, ground_plane_, image, model, camera_to_base);
+
+  pcl_ros::transformPointCloud(projected_pc, projected_pc, base_to_odom);
+  MapUtils::projectTo2D(projected_pc);
+  if (static_cast<int>(camera) == 0)
   {
-    ROS_WARN_STREAM_THROTTLE(1, "camera model not initialized or not using ground filter");
-    return;
+    MapUtils::debugPublishPointCloud(camera_projection_pub_l_, projected_pc, pcl_conversions::toPCL(stamp), "/odom",
+                                     debug_);
+    sensor_msgs::ImagePtr msg = cv_bridge::CvImage(std_msgs::Header(), "mono8", image).toImageMsg();
+    msg->header.stamp = ros::Time::now();
+    msg->header.frame_id = "/left_cam_optical";
+    process_image_pub_.publish(msg);
+  }
+  else if (static_cast<int>(camera) == 1)
```

Comment:
why do the center and right cameras not have:
```C++
    sensor_msgs::ImagePtr msg = cv_bridge::CvImage(std_msgs::Header(), "mono8", image).toImageMsg();
    msg->header.stamp = ros::Time::now();
    msg->header.frame_id = "/left_cam_optical";
    process_image_pub_.publish(msg);
```

Reply:
oh rip I forgot to remove this debugging call